src/lib/token_list_gen.ts

```diff
@@ -64,9 +64,19 @@ export const generateTokenList = async (
 
   
   const l1TokenAddresses = tokens.map((token:GraphTokenResult) => token.l1TokenAddr);
-  const l2Addresses = await getL2TokenAddresses(l1TokenAddresses, bridge);
-  
-  const tokenData = await getL2TokenData(l2Addresses, bridge);
+  const l2AddressesFromL1 = await getL2TokenAddressesFromL1(l1TokenAddresses, bridge);
+  const l2AddressesFromL2 = await getL2TokenAddressesFromL2(l1TokenAddresses, bridge);
+
+  // if the l2 route hasn't been updated yet we remove the token from the bridged tokens
+  for (let index = 0; index < l2AddressesFromL1.length; index++) {
```

Comment:
This is purely just code aesthetics, but we do this with a `tokens.filter` ?

Reply:
Switched to filter